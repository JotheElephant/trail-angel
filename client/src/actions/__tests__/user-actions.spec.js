'use strict';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockAsyncStorage = require('mock-async-storage');

const middlewares = [thunk];

import userActions from '../user-actions';
import profileData from '../../../__tests__/fixtures/profile-data';

const mockStore = configureStore(middlewares);

describe('user actions', () => {
  beforeEach(() => {
    mockAsyncStorage.mock();
  });

  afterEach(() => {
    mockStore.clearActions();
    mockAsyncStorage.release();
  });

  it('should dispatch login user action', () => {
    return mockStore.dispatch(userActions.loginUser(profileData))
      .then((result) => {
        expect(result).toMatchSnapshot();
      });
  });

  it('should dispatch logout user action', () => {
    return mockStore.dispatch(userActions.logoutUser())
      .then((result) => {
        expect(result).toMatchSnapshot();
      });
  });

  it('should receive user data', () => {
    expect(mockStore.dispatch(userActions.receiveUserData(profileData)))
      .toMatchSnapshot();
  });
});