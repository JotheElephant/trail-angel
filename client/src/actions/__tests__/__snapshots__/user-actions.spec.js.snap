exports[`user actions should log a user in 1`] = `
Object {
  "profile": Object {
    "avatarUrl": "https://lh6.googleusercontent.com/-PHUnYsSh0kM/AAAAAAAAAAI/AAAAAAAAAQQ/Y5SZ33EhyYo/photo.jpg",
    "email": "wyatt.b.lindsey@gmail.com",
    "nickname": "wyatt.b.lindsey",
    "userId": "113879522060780032239",
  },
  "type": "RECEIVE_USER_DATA",
}
`;

exports[`user actions should logout a user 1`] = `
Object {
  "type": "LOGOUT_USER",
}
`;

exports[`user actions should receive user data 1`] = `
Object {
  "profile": Object {
    "avatarUrl": "https://lh6.googleusercontent.com/-PHUnYsSh0kM/AAAAAAAAAAI/AAAAAAAAAQQ/Y5SZ33EhyYo/photo.jpg",
    "email": "wyatt.b.lindsey@gmail.com",
    "nickname": "wyatt.b.lindsey",
    "userId": "113879522060780032239",
  },
  "type": "RECEIVE_USER_DATA",
}
`;
